pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -122,13 +124,13 @@ private synchronized SegmentCompletionFSM lookupOrCreateFsm(final LLCSegmentName
           // Also good for synchronization, because it is possible that multiple threads take this path, and we don't want
           // multiple instances of the FSM to be created for the same commit sequence at the same time.
           final long endOffset = segmentMetadata.getEndOffset();
-          fsm = SegmentCompletionFSM.fsmInCommit(_segmentManager, this, segmentName, segmentMetadata.getNumReplicas(), endOffset);
+          fsm = SegmentCompletionFSM.fsmInCommit(_segmentManager, this, segmentName, segmentMetadata.getNumReplicas(), endOffset, _controllerConf.getAcceptSplitCommit());
```

Comment:
Dont add argument to every call. Instead, add a getter in SegmentCompletionManager boolean  shouldAcceptSplitCommit()